addon/-private/system/is-array-like.ts

```diff
@@ -11,7 +16,7 @@ import EmberArray from '@ember/array';
   https://github.com/emberjs/ember.js/pull/11463 `Ember.isArray` is an alias of
   `Array.isArray` hence removing the "array-like" part.
  */
-export default function isArrayLike(obj) {
+export default function isArrayLike(obj?: MaybeArrayLike) {
```

Comment:
@chriskrycho / @mike-north / @dfreeman not really sure how to type this one.

Also, side note - the check for setInterval is strange to me. Are we passing window at some point to this function??

Reply:
Correct: the point of `unknown` is that you have no idea what is passed in. You can get safer internally by way of [type narrowing](http://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types).

Reply:
this is probably best expressed as a user-defined type guard
```ts
function isArrayLike(obj: unknown): obj is ArrayLike
```

Reply:
Luckily this is a moot issue given https://github.com/emberjs/data/pull/5724